### DISCUSSION TITLE
Git artifact using GitHub App Credentials

### DISCUSSION THREAD

**Discussion by mike-donlon** [argoproj/argo-workflows#11028] [Q&A]
I am not finding any documentation how I can use a git artifact that authenticates using GitHub App Credentials in a workflow. Is it possible?

**Comment by oisin88** [comment]
I am also seeing this issue, I would like to reuse the Github App being used by ArgoCd

**Comment by wesleyscholl** [comment]
@mike-donlon @oisin88 

As of December 2024, Argo Workflows does not natively support using GitHub App credentials for authenticating Git artifacts within workflows. Available authentication methods:

- Username and Password - Utilizing personal access tokens as passwords.
- SSH Keys - Using SSH keys for authentication.

A feature request for GitHub App authentication with artifacts was opened in October 2023, #13825. As of today, this feature has not been implemented.

**Resources**

https://github.com/argoproj/argo-workflows/issues/13825

https://argo-workflows.readthedocs.io/en/latest/fields/#gitartifact
